File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) 
    {
        sort(nums.begin(), nums.end());
        int n = nums.size(), ans = INT_MIN;
        for(int i=0;i<n/2;i++) ans = max(nums[i]+nums[n-1-i],ans);
        
        return ans;
    }
};

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution(object):
    def minPairSum(self, nums):
        nums.sort()
        n = len(nums)
        ans = float('-inf')
        
        for i in range(n // 2):
            ans = max(nums[i] + nums[n - 1 - i], ans)
            
        return ans
